src/chain/validators-impl.ts

```diff
@@ -333,6 +333,10 @@ export class ValidatorsImpl<Chain> implements Validators<Chain> {
     return this.addStandardValidation(validator.isWhitelisted, chars);
   }
   matches(pattern: RegExp | string, modifiers?: string) {
-    return this.addStandardValidation(validator.matches, pattern, modifiers);
+    return this.addStandardValidation(
+      validator.matches,
+      typeof pattern === 'string' ? pattern : pattern.source,
+      modifiers,
```

Comment:
Isn't this counter-intuitive?
I mean this means that a user specifying a regex object with flags need to also set the optional modifiers separately
e.g.
```js 
  .matches(/foo/g) // => becomes /foo/
  .matches(/foo/g, 'g') // => is correctly /foo/g
```

Reply:
Hi, yes, I forgot about the flags...I'll fix it tomorrow...